src/win32/kerberos_win32.cc

```diff
@@ -156,21 +156,22 @@ NAN_METHOD(InitializeClient) {
 
     KerberosWorker::Run(callback, "kerberos:InitializeClient", [=](KerberosWorker::SetOnFinishedHandler onFinished) {
         sspi_client_state* client_state = sspi_client_state_new();
-        std::shared_ptr<sspi_result> result(auth_sspi_client_init(
+        sspi_result result = auth_sspi_client_init(
             (WCHAR*)service.c_str(), gss_flags, (WCHAR*)user.c_str(), user.length(),
             (WCHAR*)domain.c_str(), domain.length(), (WCHAR*)password.c_str(), password.length(),
-            (WCHAR*)mech_oid.c_str(), client_state), ResultDeleter);
+            (WCHAR*)mech_oid.c_str(), client_state);
 
+        // XXX a) The comment is wrong; b) This leaks memory
```

Comment:
Go ahead and remove, can you point out the leak just for my knowledge? or it did leak before this change because it didn't free the pointer?

Reply:
> or it did leak before this change because it didn't free the pointer?

It leaked because we didn’t call `auth_sspi_client_clean()` at that point, and only freed the pointer instead of both the pointer and the struct’s contents.